src/shisen_cpp/camera/provider/image_provider.cpp

```diff
@@ -68,13 +68,8 @@ void ImageProvider::update_mat()
 void ImageProvider::set_mat(cv::Mat mat)
 {
   current_mat_image = mat;
-
-  // Set image according to the compression quality
-  if (compression_quality > 0) {
-    set_image(current_mat_image.compress(compression_quality));
-  } else {
-    set_image(current_mat_image);
-  }
+  set_image(*(cv_bridge::CvImage(std_msgs::msg::Header(), "bgr8", mat)
+                  .toImageMsg()));
```

Comment:
I suggest to break this statement into several lines, e.g.:

```c++
const auto foo = make_foo(something, something_else);
set_image(foo.do_omething());
```

Reply:
ok